rescan.go

```diff
@@ -508,55 +474,54 @@ func rescan(chain ChainSource, options ...RescanOption) error {
 		// notification, then we can re-process it without any issues.
 		nextBlockHeight := uint32(curStamp.Height + 1)
 		_, err := chain.GetFilterHeaderByHeight(nextBlockHeight)
-		if header.BlockHash() != curStamp.Hash && err != nil {
+		if !isRetry && err != nil {
```

Comment:
Why is it acceptable to not be able to fetch a filter header if this is  a block that we're retrying? Shouldn't we immediately fall back to fetching this filter again?

Reply:
I'm not sure what this check achieves in the first place. I noticed it was reverted as part of https://github.com/lightninglabs/neutrino/pull/104, but I don't know if that still applies cc @halseth.

The change as part of this diff is just maintaining the same check, i.e. `(header.BlockHash() != curStamp.Hash) == !isRetry`.